gulp-npmworkspace/src/Plugins/WorkspacePackages.ts

```diff
@@ -78,13 +85,19 @@ function collectPackages(context: PackageDependencyContext): TransformAction {
 
         context.addPackage(packageDescriptor, file);
 
-        let packageDependencies: IDictionary<string>
-            = _.extend({ }, packageDescriptor.dependencies, packageDescriptor.peerDependencies, packageDescriptor.devDependencies, packageDescriptor.optionalDependencies);
-
-        for (let packageName in packageDependencies) {
+        for (let packageName in packageDescriptor.dependencies) {
             context.addPackageDependency(packageDescriptor, packageName);
         }
 
+        if ( ! productionOnly ) {
+            let extendedDependencies: IDictionary<string>
+                = _.extend({ }, packageDescriptor.peerDependencies, packageDescriptor.devDependencies, packageDescriptor.optionalDependencies);
+
+            for (let packageName in packageDependencies) {
```

Comment:
Compilation error on line 96:
`packageDependencies` is unknown. Should this be `extendedDependencies` from the line above?

I'll fix it in master since I've used the auto merge feature.